STM32/AC/HeatCtrl/Src/HeatCtrl.c

```diff
@@ -128,3 +128,24 @@ void setPWMPin(int pin, int pwmPct, int duration_ms)
         ctx->periodBegin = HAL_GetTick();
     }
 }
+
+void adjustPWMDown()
+{
+    for(HeatCtrl *ctx = heaters; ctx < &heaters[noOfHeaters]; ctx++)
+    {
+    	if (ctx->pwmPercent >= 1)
+    	{
+    		ctx->pwmPercent -= 1;
+    		ctx->pwmDuration = MAX_DURATION;
```

Comment:
does this value has any relation to the on duration the LoopControl sends?

Reply:
Yes and no. 'pwmDuration' is the amount of ms the pwm as a whole should run. The switching on/off comes from pwmPercent and pwmPeriod. (line 63 of HeatCtrl.c). When setting pwmDuration to max it keeps running the pwm signal indefinitely with the same frequency as set from LoopControl. 

If we do not set pwmDuration it will shut off after the last period is sent by LoopControl and therefore not stay on the output that gives the maxTemperature.